codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -603,10 +609,49 @@ impl Field {
         // TODO: should we dis-allow indexed fields that are either `id` or derived From fields?
         let is_indexed = indexed_count > 0;
 
-        let field_type =
+        let mut field_type =
             FieldType::from_obj_field_type(&field.field_type, derived_from_field, is_indexed)
                 .context(format!("Failed parsing field {}", field.name))?;
 
+        if numeric_directives.len() == 1 {
+            if let FieldType::RegularField {
+                field_type: UserDefinedFieldType::NonNullType(inner),
+                ..
+            } = &mut field_type
+            {
+                if let UserDefinedFieldType::Single(GqlScalar::BigInt(None)) = inner.as_mut() {
+                    if let Some(directive) = numeric_directives.get(0) {
+                        let precision = directive.arguments.iter().find_map(|(k, v)| {
+                            if k == "precision" {
+                                if let Value::Int(i) = v {
+                                    Some(i.as_i64().unwrap() as u32)
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        });
+                        let scale = directive.arguments.iter().find_map(|(k, v)| {
+                            if k == "scale" {
+                                if let Value::Int(i) = v {
+                                    Some(i.as_i64().unwrap() as u32)
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        });
+                        *inner = Box::new(UserDefinedFieldType::Single(GqlScalar::BigInt(Some((
+                            precision.unwrap_or(0),
+                            scale.unwrap_or(0),
+                        )))));
+                    }
+                }
+            }
+        }
```

Comment:
No user feedback implemented yet ( Such as writing a message to the user if they put the precision directive on a non-int field.)

Reply:
Hey @JasoonS, I have a number of thoughts on this parsing implementation. Could you clarify:

1. Should the user always provide both scale AND precision with the numeric directive?
2. Do we want to be able to manually tune this more than a fixed recommended size for the given use case? Or will we always essentially be using the same values for scale and precision?

Reply:
Ok cool. So precision should not be optional with the directive but scale should be optional?

Or should it be scale be non optional for the case of big decimal?

What I'm trying to understand by number 2 is could we not introduce a fixed type for the use case like Numeric72 or whatever number precision we need? Or is it likely that we will have different needs for precision and scaled depending on each individual use case?

Reply:
Ok, gotcha - for 2 -  I think that it is quite simple for us to make it completely customizable and I think it is the edge case that a user will want to set this, so I think adding extra types is maybe not the best idea :thinking: 

For 1: I wrote a proposal on Discord. I'll keep the conversation here instead. I think that we can just not use scale at all for BigInt, because scale, as far as I understand, shouldn't be necessary for BigInt and should only be something users would care about modifying for BigDecimal.

 CurrentProposal:
 ```graphql
type Entity {
  id: ID!
  exampleBigInt: BigInt @decimalPrecision(76)
  exampleBigDecimal: BigDecimal @numeric(precision: 76, scale: 5)
}
```